Myth7x/xp-discord-theme
# Editing my themes efficiently

Hi there! Just wanted to let you know that there's a much easier way of making changes to my theme, rather than editing the human-unfriendly `xp.min.css` file.

# Contents of this pull request
- The changes that you made, added into the SCSS files. These are marked by `// START Edits by Myth7x` and `// END Edits by Myth7x`.
- Some updates from the main branch of my theme.
- A new GitHub Actions workflow, which automatically compiles SCSS to CSS and deploys to GitHub Pages. This makes the `@import` in the `Exponent.theme.css` work correctly.

# Making changes, testing, and deploying
1. Install [pnpm](https://pnpm.io/installation)
2. In the terminal, go to the `xp-discord-theme` folder and run `pnpm run dev`.
    - This compiles `xp.theme.css` to your BetterDiscord themes folder, every time a change is detected in the `xp-discord-theme/scss` folder. 
    - No need to push changes and wait for GitHub to update, just see changes directly as you work.
4. Make your desired changes to the files in `xp-discord-theme/scss` folder.
    - The files are split up in what I hope is a logical manner, and comments explain what the selectors target.
    - [SCSS](https://sass-lang.com/documentation/) has similar syntax to CSS.
6. Once you're happy with your changes, commit and push to your `main` branch.
    - This will trigger the GitHub workflow to compile the SCSS to CSS and create/update `https://myth7x.github.io/xp-discord-theme/xp.min.css`. Once the workflow has completed, you'll see your changes when loading `Exponent.theme.css`.

If there's anything else I can do to help, let me know!
